tests/dragonfly/snapshot_test.py

```diff
@@ -356,3 +357,20 @@ async def info_in_loop():
     await save_task
     save_finished = True
     await info_task
+
+
+@dfly_args({**BASIC_ARGS, "dbfilename": "test-bgsave"})
+async def test_bgsave_and_save(async_client: aioredis.Redis):
+    await async_client.execute_command("DEBUG POPULATE 20000")
+
+    await async_client.execute_command("BGSAVE")
+    with pytest.raises(redis.exceptions.ResponseError):
+        await async_client.execute_command("BGSAVE")
+
+    await asyncio.sleep(2)
```

Comment:
instead of sleep for 2 seconds I suggest use

async def is_saving():
        return "saving:1" in (await async_client.execute_command("INFO PERSISTENCE"))

while not await is_saving():  # wait for server start saving
        asyncio.sleep(0.1)